lib/util/statsdAdapter.ts

```diff
@@ -0,0 +1,64 @@
+import * as os from "os";
+import { Configuration } from "../configuration";
+
+import {ClientOptions, StatsD} from "hot-shots"
+
+export type StatsdAdapterStat = string | string[];
+export type StatsdAdapterTags = { [key: string]: string } | string[];
+export type StatsdAdapterCallback = (err?: Error) => void;
+export interface StatsdAdapterOptions {
+    prefix?: string;
+    suffix?: string;
+    globalTags?: string[];
+
+    // This is to support client library specific options
+    additionalOptions?: {
+        [key: string]: string;
+    };
+}
+
+export interface StatsdAdapter {
+    increment(stat: StatsdAdapterStat, value: number, sampleRate?: number, tags?: StatsdAdapterTags, callback?: StatsdAdapterCallback): void;
+    timing(stat: StatsdAdapterStat, value: number, sampleRate?: number, tags?: StatsdAdapterTags, callback?: StatsdAdapterCallback): void;
+    gauge(stat: StatsdAdapterStat, value: number, sampleRate?: number, tags?: StatsdAdapterTags, callback?: StatsdAdapterCallback): void;
+
+    // TODO: This one is dogstatsd specific, as such it should be removed or made optional
+    event(stat: StatsdAdapterStat, text?: string, options?: {}, tags?: StatsdAdapterTags, callback?: StatsdAdapterCallback): void;
```

Comment:
You are using dogstatsd events to track registrations, shutdowns and operational failures. This may require having a separate adapter going forward, to cleanly separate concerns and allow users to setup their own event ingression system.